src/Engine/Manager/ScriptManager.cpp

```diff
@@ -424,12 +429,113 @@ void ScriptManager::BuildAllScripts() {
     }
 }
 
+void ScriptManager::FillPropertyMap(Script* script) {
+
+    BuildScript(script->scriptFile);
+    CreateInstance(script);
+
+    int propertyCount = script->instance->GetPropertyCount();
+
+    for (int n = 0; n < propertyCount; n++) {
+
+        int typeId = script->instance->GetPropertyTypeId(n);
+        void *varPointer = script->instance->GetAddressOfProperty(n);
+
+        std::map<std::string, std::map<int, void*>>::iterator it = script->propertyMap.find(script->instance->GetPropertyName(n));
+        if (it != script->propertyMap.end()) {
+
+            std::map<int, void*>::iterator it2 = script->propertyMap[script->instance->GetPropertyName(n)].find(typeId);
```

Comment:
You have just acquired an iterator. Instead of searching `propertyMap` again, you could just deref the iterator to get the inner map you are looking for (same applies on the line below).